src/Task/PhpCsFixerV2.php

```diff
@@ -72,7 +72,15 @@ public function run(ContextInterface $context)
         $arguments->addOptionalArgument('--allow-risky=%s', $config['allow_risky'] ? 'yes' : 'no');
         $arguments->addOptionalArgument('--cache-file=%s', $config['cache_file']);
         $arguments->addOptionalArgument('--config=%s', $config['config']);
-        $arguments->addOptionalCommaSeparatedArgument('--rules=%s', $config['rules']);
+
+        $rules = $config['rules'];
+        // JSON-encode rules if specified as a map.
+        if (array_values($rules) !== $rules) {
+            $arguments->addOptionalArgument('--rules=%s', json_encode($rules));
+        } else {
```

Comment:
It's all about the mental map of the code. In one look, you can tell that if you got rules, you want to add them as an argument. Independently of the value, you know that you want to add an additional parameter. There is one single path, one single responsibility of the if statement.
Agree: for the actual value, you'll need the ternary statement. But that is just one line of logic.

However, If you use the else statement. You'll have to check 2 paths for wath the code is doing + you must find out what the 2 argument functions are doing exactly. In your case, you'll also have to add an additional if or ternary statement to check if the array is empty sinct the json function will always create the string '[]';